src/dsp_tools/utils/xmlupload/project_client.py

```diff
@@ -37,67 +38,74 @@ def get_ontology_iri_dict(self) -> dict[str, str]:
 
     def get_project_iri(self) -> str:
         """Get the IRI of the project to which the data is being uploaded."""
         if not self.project_info:
-            self.project_info = _get_project_info_from_server(self.server, self.shortcode)
+            self.project_info = _get_project_info_from_server(self.con, self.shortcode)
         return self.project_info.project_iri
 
     def get_ontology_iris(self) -> list[str]:
         """Get the ontology IRIs of the project to which the data is being uploaded."""
         if not self.project_info:
-            self.project_info = _get_project_info_from_server(self.server, self.shortcode)
+            self.project_info = _get_project_info_from_server(self.con, self.shortcode)
         return self.project_info.ontology_iris
 
     def get_ontology_name_dict(self) -> dict[str, str]:
         """Returns a mapping of ontology names to ontology IRIs."""
         if not self.project_info:
-            self.project_info = _get_project_info_from_server(self.server, self.shortcode)
+            self.project_info = _get_project_info_from_server(self.con, self.shortcode)
         return {_extract_name_from_onto_iri(iri): iri for iri in self.project_info.ontology_iris}
 
     def get_ontology_iri_dict(self) -> dict[str, str]:
         """Returns a mapping of ontology IRIs to ontology names."""
         if not self.project_info:
-            self.project_info = _get_project_info_from_server(self.server, self.server)
+            self.project_info = _get_project_info_from_server(self.con, self.shortcode)
         return {iri: _extract_name_from_onto_iri(iri) for iri in self.project_info.ontology_iris}
 
 
-def _get_project_info_from_server(server: str, shortcode: str) -> ProjectInfo:
-    project_iri = _get_project_iri_from_server(server, shortcode)
-    ontologies = _get_ontologies_from_server(server, project_iri)
+def _get_project_info_from_server(con: Connection, shortcode: str) -> ProjectInfo:
+    project_iri = _get_project_iri_from_server(con, shortcode)
+    ontologies = _get_ontologies_from_server(con, project_iri)
     return ProjectInfo(project_iri=project_iri, ontology_iris=ontologies)
 
 
-def _get_project_iri_from_server(server: str, shortcode: str) -> str:
-    url = f"{server}/admin/projects/shortcode/{shortcode}"
-    res: Response = try_network_action(requests.get, url=url, timeout=5)
-    if res.status_code != 200:
-        raise UserError(f"A project with shortcode {shortcode} could not be found on the DSP server")
-    iri: str = res.json()["project"]["id"]
+def _get_project_iri_from_server(con: Connection, shortcode: str) -> str:
+    try:
+        url = f"/admin/projects/shortcode/{shortcode}"
+        res: dict[str, Any] = try_network_action(con.get, route=url)
+        iri: str = res["project"]["id"]
+    except BaseError as e:
+        logger.exception(e)
+        raise BaseError(f"A project with shortcode {shortcode} could not be found on the DSP server") from None
```

Comment:
To me it does not make sense to put instructions to the user into exceptions. I would much prefer having a clearly defined set of exceptions modelling the issue, so that we can separate the exception message which should be diagnostic information in the logs, from the user feedback we want to print in the console.  
I suggest I change this to a UserError and we keep the question of exception messages for a separate discussion.

Reply:
Okay